### PR TITLE
Security / JWT

### PR DESCRIPTION
**Pull request by aksoli666**
+Added the following method to the AuthenticationController: public UserLoginResponseDto login(UserLoginRequestDto request) 
+Created AuthenticationService, JwtAuthenticationFilter, JwtUtil classes
+ /auth/login public available 
+Used new filter in the SecurityFilterChain bean
+Used data validation in the new DTO classes

### CODE REVIEW

File: src/main/java/online/book/store/filter/JwtAuthFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(auth);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getTokenFromRequest(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (kerrrusha) [inline src/main/java/online/book/store/filter/JwtAuthFilter.java:45]: magic number, just use length of `"Bearer "` constant

File: src/main/java/online/book/store/filter/JwtAuthFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication auth = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(auth);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getTokenFromRequest(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
```
Comment (kerrrusha) [inline src/main/java/online/book/store/filter/JwtAuthFilter.java:44]: should be constants

File: src/main/java/online/book/store/OnlineBookStoreApplication.java [hunk 3]
```diff
@@ -9,5 +9,6 @@
 public class OnlineBookStoreApplication {
     public static void main(String[] args) {
         SpringApplication.run(OnlineBookStoreApplication.class, args);
+        System.out.println("OnlineBookStoreApplication started");
```
Comment (kerrrusha) [inline src/main/java/online/book/store/OnlineBookStoreApplication.java:12]: spring already prints that, moreover, for logging its better to use `@Slf4j` annotation and `log.info("your message")`
```suggestion
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/online/book/store/filter/JwtAuthFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthFilter extends OncePerRequestFilter {
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String BEARER_PREFIX = "Bearer ";
+    private static final int BEARER_PREFIX_LENGTH = BEARER_PREFIX.length();
```
Comment (okuzan) [inline src/main/java/online/book/store/filter/JwtAuthFilter.java:23]: no need to make it a constant too, can use it in-place :)

**Review by okuzan — APPROVED** [review]

**Pull request merged**